native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/NativeImagePlugin.java

```diff
@@ -348,6 +350,48 @@ private void configureJvmReachabilityConfigurationDirectories(Project project, G
+        GraalVMReachabilityMetadataRepositoryExtension repositoryExtension = reachabilityExtensionOn(graalExtension);
+        Provider<GraalVMReachabilityMetadataService> serviceProvider = project.getGradle()
+                .getSharedServices()
+                .registerIfAbsent("nativeConfigurationService", GraalVMReachabilityMetadataService.class, spec -> {
+                    LogLevel logLevel = determineLogLevel();
+                    spec.getParameters().getLogLevel().set(logLevel);
+                    spec.getParameters().getUri().set(repositoryExtension.getUri());
+                    spec.getParameters().getCacheDir().set(new File(project.getGradle().getGradleUserHomeDir(), "native-build-tools/repositories"));
+                });
+        options.getExcludeConfigArgs().addAll(repositoryExtension.getEnabled().flatMap(enabled -> {
+            if (enabled) {
+                if (repositoryExtension.getUri().isPresent()) {
+                    Configuration classpath = project.getConfigurations().getByName(sourceSet.getRuntimeClasspathConfigurationName());
+                    Set<String> excludedModules = repositoryExtension.getExcludedModules().getOrElse(Collections.emptySet());
+                    Map<String, String> forcedVersions = repositoryExtension.getModuleToConfigVersion().getOrElse(Collections.emptyMap());
+                    return serviceProvider.map(repo -> classpath.getIncoming().getResolutionResult().getAllComponents().stream().flatMap(component -> {
+                        ModuleVersionIdentifier moduleVersion = component.getModuleVersion();
+                        return repo.findConfigurationsFor(query -> {
+                                    String module = Objects.requireNonNull(moduleVersion).getGroup() + ":" + moduleVersion.getName();
+                                    if (!excludedModules.contains(module)) {
+                                        query.forArtifact(artifact -> {
+                                            artifact.gav(module + ":" + moduleVersion.getVersion());
+                                            if (forcedVersions.containsKey(module)) {
+                                                artifact.forceConfigVersion(forcedVersions.get(module));
+                                            }
+                                        });
+                                    }
+                                    query.useLatestConfigWhenVersionIsUntested();
+                                }).stream()
+                                .filter(org.graalvm.reachability.Configuration::isOverride)
+                                .flatMap(configuration -> Stream.of(
+                                        "--exclude-config",
+                                        "/" + moduleVersion.getName() + "-" + moduleVersion.getVersion() + "\\.jar$",
```

Comment:
This is quite fragile, but this has to be "fixed' somehow in `native-image`: relying on file names is not great. This would break for classifiers for example, or even for different variants of a single module (e.g guava).

Reply:
On Maven side, I think I retrieve the exact dependency file name associated with an artifact where the override flag is true, do we have a way to do the same on Gradle side?

Reply:
@sdeleuze Gradle supports excluding using GAV coordinates:
```
excludeConfig.put("org.example.test:artifact:version", ["META-INF/native-image/*", "config/*"]) // Excludes configuration that matches one of given regexes from JAR of dependency with said coordinates.
```

So you might be able to utilize this :)

Reply:
Very good point, thanks for sharing @lazar-mitrovic, I will update the PR accordingly.

Reply:
Ok so I did the change in 8d9478cd0c003a3e43fc6a0a29f367a328065075, exclusion are passed as expected to `native-image` but they did not work in practice because `excludeConfig` is broken. I fixed it via f8dfca52a059d28fe761c117be36fb9556f7bac9 and now it works as expected.